packages/ui-extensions/src/surfaces/point-of-sale/components/CameraScanner/CameraScanner.ts

```diff
@@ -9,21 +9,52 @@ import {BannerVariant} from '../Banner/Banner';
 export interface CameraScannerBannerProps {
   /**
    * The title of the banner.
+   * @deprecated CameraScannerBannerProps has been deprecated. Use the CameraScannerNextProps instead.
    */
   title: string;
 
   /**
    * The appearance of the banner.
+   * @deprecated CameraScannerBannerProps has been deprecated. Use the CameraScannerNextProps instead.
    */
   variant: BannerVariant;
+
   /**
    * The visibility state of the banner.
+   * @deprecated CameraScannerBannerProps has been deprecated. Use the CameraScannerNextProps instead.
    */
   visible: boolean;
 }
 
+/**
+ * @property scanStatus - the state of scanning the barcode.
+ * @property overlayScannerText - the scanner overlay default text on top of the scanner window.
+ * @property bannerMessage - The title of the banner.
+ */
+
+export interface CameraScannerNextProps {
+  /**
+   * the state of scanning the barcode
+   */
+  scanStatus: 'error' | 'success' | 'scanning';
+
+  /**
+   * the scanner overlay default text on top of the scanner window.
+   */
+  overlayScannerText?: string;
+
+  /**
+   * The error banner message.
+   */
+  bannerMessage?: string;
```

Comment:
> The figma suggests that a success or informational banner is also possible. Why are we removing that possibility from extensions?
That is the only exception where it was allowed to keep the checkout with gift cards unchanged. In general, we do not use the informational banner. But it can be something that be addressed in later stage. It also maintains the same current behaviour in the existing scanner. It would close upon a success scan 

Why is it the extension's responsibility to manage the state of the scanner? Can't the scanner understand when it has successfully scanned something, or encounters an error?
I am confused, how will the scanner know if it was successful scan or there was an error. The user is the one that builds logic telling the scanner if that is a valid scan or invalid scan based on the scanned barcode